SubEthaEdit-Mac/PlainTextWindow.xib

```diff
@@ -24,8 +24,37 @@
                 <autoresizingMask key="autoresizingMask"/>
             </view>
             <connections>
+                <outlet property="cuationView" destination="TbK-0x-Bg8" id="Bb1-vT-g2V"/>
                 <outlet property="delegate" destination="-2" id="9"/>
             </connections>
         </window>
+        <customView id="TbK-0x-Bg8">
+            <rect key="frame" x="0.0" y="0.0" width="16" height="16"/>
+            <autoresizingMask key="autoresizingMask" flexibleMinX="YES" flexibleMaxX="YES" flexibleMinY="YES" flexibleMaxY="YES"/>
+            <subviews>
+                <imageView horizontalHuggingPriority="251" verticalHuggingPriority="251" id="dSP-f5-pqd">
+                    <rect key="frame" x="0.0" y="0.0" width="16" height="16"/>
+                    <autoresizingMask key="autoresizingMask" flexibleMinX="YES" flexibleMaxX="YES" flexibleMinY="YES" flexibleMaxY="YES"/>
+                    <imageCell key="cell" controlSize="mini" refusesFirstResponder="YES" alignment="left" imageScaling="proportionallyDown" image="CautionSymbol" id="sbd-0b-uhX"/>
+                </imageView>
+            </subviews>
+            <connections>
+                <binding destination="-2" name="hidden" keyPath="window.document.hasAlerts" id="Tbw-Nv-TTV">
```

Comment:
while seemingly neat, this just introduces a retain cycle that needs to be resolved manually (e.g. in the current state this observation prohibits any window from being deallocated). I'd rather you either explicitly call a validation method in the window, or do manual key value observing that you remove on dealloc in the window)

Reply:
Could you elaborate on this a bit more? I fixed an unrelated typo that was causing a memory leak (I was accidentally calling `[super addWindowController:]` in `removeWindowController:` https://github.com/tolmasky/SubEthaEdit/commit/637bf9ecf0b383ed13e39be7ff4476f69acad45c ), and after this I can confirm that when you close a window, the window, document, and window controller all properly dealloced. However, perhaps there is a specific situation you are running into that is causing a separate cycle.

Reply:
The contents of a object controller are held on strongly. Thusly in the nibs top level objects there exists a hierarchy between the window and the object controller. In other cases that relationship caused the cycle as can be observed by explicitly clearing out the contents of those controllers in pre-dealloc methods. But maybe in this case this is different, didn't check if you actually added the controller.

Reply:
Indeed the cycle seems to be fixed with the wrong call to super, so the binding can stay.